src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -281,7 +283,7 @@ bool TryConvertValueTuple(object value, Destructuring destructuring, Type valueT
             return false;
         }
 
-        bool TryConvertCompilerGeneratedType(object value, Destructuring destructuring, Type valueType, out LogEventPropertyValue result)
+        bool TryConvertCompilerGeneratedType(object value, Destructuring destructuring, Type valueType, [NotNullWhen(true)] out LogEventPropertyValue? result)
```

Comment:
These are nicer to consume when they're written as:

```csharp
[MaybeNullWhen(false)] out LogEventPropertyValue result
```

I.e. the out var is not nullable, and the compiler checks that the return value has been used. (This is how `Dictionary<K,V>` works when `V` is a non-nullable reference type.)

Reply:
but LogEventPropertyValue is a reference type, so it is not comparable to the Dictionary scenario?

Reply:
That's a good question; I have a couple of things to check - will tinker and report back :-) 👍

Reply:
any outcome from the tinkering?

Reply:
Sorry, hectic couple of weeks again! Starting to catch up but slow going :)

Reply:
no rush on my side, just making sure it hadnt slipped off your radar

Reply:
Thanks :-) :+1:

Reply:
Thanks for hanging in there, mate - I think your approach is the right one, squashing now!